lms/migrations/versions/f3d631c110bf_canvas_files_enabled.py

```diff
@@ -0,0 +1,81 @@
+"""
+Make canvas files enabled an AI setting.
+
+Revision ID: f3d631c110bf
+Revises: 0b7391f5b1e3
+Create Date: 2022-12-28 10:15:40.830444
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+revision = "f3d631c110bf"
+down_revision = "5f0057d6d60f"
+
+
+def upgrade():
+    """
+    Explicitly store the `files_enabled` setting for canvas instances.
+
+    Currently the logic for this "setting" is:
+
+    ```
+    enabled = (request.product.family == Product.Family.CANVAS) and (
+        "custom_canvas_course_id" in request.lti_params
```

Comment:
All lti_params with custom_canvas_course_id are canvas but not all Product.Family.CANVAS have custom_canvas_course_id.

I reckon it's a shame we can't tell what LMS an AI uses with a simple query.

Reply:
Yes. This is crazy